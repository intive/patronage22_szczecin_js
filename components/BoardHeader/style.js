import styled from 'styled-components'

import { breakpoint } from '../../config/breakpoints'

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  padding: 1.5rem 5%;
  background: #E1E1E1;
  gap: 1rem;
  padding: 0 -5%;

  @media only screen and ${breakpoint.xl} {
    padding: 1.5rem 2.5rem;
  }
`
export const TitleWrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 10px;
  justify-content: space-between;
  align-items: flex-start;
  width: 100%;
  height: fit-content;
`
export const Title = styled.h1`
  font-style: normal;
  font-weight: normal;
  font-size: 2.25rem;
  line-height: 2.5rem;
  color: #515151;
`
export const StyledLink = styled.a`
  text-decoration: none;
  width: fit-content;
  font-style: normal;
  font-weight: normal;
  font-size: .75rem;
  line-height: 1rem;
  color: #626262;
  &:hover {
    text-decoration: underline;
  }
`
