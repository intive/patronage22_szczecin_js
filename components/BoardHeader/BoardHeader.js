import Link from 'next/link'
import Button from '../Button/Button'
import { Container, TitleWrapper, Title, StyledLink } from './style'

const createNewColumn = () => {
  console.log('teraz powstanie nowa kolumna')
}

const BoardHeader = (props) => {
  return (
    <Container>
      <Link href='/' passHref><StyledLink>{props.returnLinkText}</StyledLink></Link>
      <TitleWrapper>
        <Title>{props.title}</Title>
        <Button onClick={createNewColumn}>{props.buttonText}</Button>
      </TitleWrapper>
    </Container>
  )
}

export default BoardHeader
