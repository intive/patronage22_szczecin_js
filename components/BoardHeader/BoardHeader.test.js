/* globals describe, expect, it */

import { render } from '@testing-library/react'
import BoardHeader from './BoardHeader'

describe('BoardHeader', () => {
  it('should render BoardHeader', () => {
    const { container } = render(<BoardHeader returnLinkText='Return to main board' buttonText='New Column' title='Board name 1' />)
    expect(container).toMatchSnapshot()
  })
})
