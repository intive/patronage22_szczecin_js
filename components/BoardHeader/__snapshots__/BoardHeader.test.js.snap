// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardHeader should render BoardHeader 1`] = `
<div>
  <section
    class="sc-dkPtRN hJxNlN"
  >
    <a
      class="sc-jRQBWg jfsFve"
      href="/"
    >
      Return to main board
    </a>
    <div
      class="sc-hKwDye AeogP"
    >
      <h1
        class="sc-eCImPb cIClSu"
      >
        Board name 1
      </h1>
      <button
        class="sc-bdvvtL dRBcfm"
      >
        New Column
      </button>
    </div>
  </section>
</div>
`;
